subcmd/edit.go

```diff
@@ -0,0 +1,218 @@
+		return false, errors.New("could not read config.toml: " + err.Error())
+	}
+
+	viitor, err := cmd.identifyEditor(cfg)
+	if err != nil || viitor == "" {
+		//FIXME: Which error code to use?
+		return false, &Error{Code: 30, Msg: "No usable viitor found"}
+	}
+
+	changeWasMade := false
+	//FIXME: Run single vim instance? Leads to problems if the configured
+	//editor does not support to open multiple files
+	for _, reposPath := range reposPathList {
+
+		// Edit plugconf file
+		plugconfPath := reposPath.Plugconf()
+
+		// Install a new template if none exists
+		if !pathutil.Exists(plugconfPath) {
+			getCmd := new(getCmd)
+			logger.Debugf("Installing new plugconf for '%s'.", reposPath)
+			getCmd.downloadPlugconf(reposPath)
+		}
+
+		// Remember modification time before opening the editor
+		info, err := os.Stat(plugconfPath)
+		if err != nil {
+			return false, err
+		}
+		mTimeBefore := info.ModTime()
+
+		// Call the editor with the plugconf file
+		vimCmd := exec.Command(viitor, plugconfPath)
+		vimCmd.Stdin = os.Stdin
+		vimCmd.Stdout = os.Stdout
+		if err = vimCmd.Run(); err != nil {
+			//FIXME: Don't abort immediately, but try to edit remaining files?
+			return false, err
+		}
+
+		// Get modification time after closing the editor
+		info, err = os.Stat(plugconfPath)
+		if err != nil {
+			return false, err
+		}
+		mTimeAfter := info.ModTime()
+
+		// A change was made if the modification time was updated
+		changeWasMade = changeWasMade || mTimeAfter.After(mTimeBefore)
+
+		// Remove repository from lock.json
+		err = lockJSON.Repos.RemoveAllReposPath(reposPath)
+		err2 := lockJSON.Profiles.RemoveAllReposPath(reposPath)
+		if err == nil || err2 == nil {
+			// ignore?
+		}
+	}
+
+	// Write to lock.json
+	if err = lockJSON.Write(); err != nil {
+		return changeWasMade, err
+	}
+	return changeWasMade, nil
+}
+
+func (cmd *editCmd) parseArgs(args []string) (pathutil.ReposPathList, error) {
+	fs := cmd.FlagSet()
+	fs.Parse(args)
+	if cmd.helped {
+		return nil, ErrShowedHelp
+	}
+
+	if len(fs.Args()) == 0 {
+		fs.Usage()
+		return nil, errors.New("repository was not given")
+	}
+
+	// Normalize repos path
+	reposPathList := make(pathutil.ReposPathList, 0, len(fs.Args()))
+	for _, arg := range fs.Args() {
+		reposPath, err := pathutil.NormalizeRepos(arg)
+		if err != nil {
+			return nil, err
+		}
+		reposPathList = append(reposPathList, reposPath)
+	}
+
+	return reposPathList, nil
+}
+
+func (cmd *editCmd) identifyEditor(cfg *config.Config) (string, error) {
+	var editors []string
+
+	// if an editor is specified as commandline argument, consider it
+	// as alternative
+	if cmd.editor != "" {
+		editors = append(editors, cmd.editor)
+	}
+
+	// if an editor is configured in the config.toml, consider it as
+	// alternative
+	if cfg.Edit.Editor != "" {
+		editors = append(editors, cfg.Edit.Editor)
+	}
+
+	// specifiy a fixed list of other alternatives
+	editors = append(editors, "$VISUAL", "nvim", "vim", "sensible-editor", "$EDITOR")
```

Comment:
Please change the priorities to 

1. `pathutil.VimExecutable()`
1. `sensible-editor`
1. `$VISUAL`
1. `$EDITOR`

But current `pathutil.VimExecutable()` doesn't look up `nvim` executable path.
It is another problem though. #243

Reply:
Are you sure you want to prefer `sensible-editor` over `$VISUAL`? I think an environment variable should override a "fixed" editor.

Reply:
Ah sorry, you are right.